portfolio/src/main/webapp/index.html

```diff
@@ -45,7 +45,7 @@ <h2>Comments</h2>
             <option value="👽">👽</option>
           </select>
           <br></br>
-          <input type="submit"/>
+          <input type="button" value="Submit" onclick="addComment()"/>
```

Comment:
You can keep the submit button and listen to the onSubmit on the form (https://www.w3schools.com/jsref/event_onsubmit.asp)

This has the advantage, that it still works, even if the user has JavaScript disabled - it would then submit the form without JavaScript and with a full page redirect.

Reply:
I tried it with onsubmit first, but it seems to still automatically redirect and reload the the page. I can still use onsubmit if I do I use `onsubmit="addComment();return false"` to stop default behaviour, but in this case the form also doesn't work if JavaScript is disabled.

Reply:
One idea is to use `onsubmit="return addComment()"` and then you can control in your addComment function if you want to stop the default behavior by returning false. See https://stackoverflow.com/questions/16262797/html-form-action-and-onsubmit-issues

Another advantage is that you can specify "required" in some of your inputs and this basic validation will automatically be done if the user clicks submit, but it will be bypassed when using onClick on a random button.

Reply:
For some reason returning false in the function also didn't work, so I went with `addComment();return false`. The required option is useful!